src/fileimpl.cpp

```diff
@@ -409,6 +409,24 @@ class Grouping
     return mp_pathDirentAccessor->getDirent(idx);
   }
 
+  FileImpl::FindxResult FileImpl::findxMetadata(const std::string& name) const {
+    auto r = findx('M', name);
+    if (!r.first) {
+      return r;
+    }
+    auto dirent_idx = r.second;
+    auto dirent = getDirent(dirent_idx);
+    auto watchdog = 50U;
+    while (dirent->isRedirect() && --watchdog) {
+      dirent_idx = dirent->getRedirectIndex();
+      dirent = getDirent(dirent_idx);
+    }
+    if (watchdog) {
+      return {true, dirent_idx};
+    }
+    return {false, entry_index_t(0)};
+  }
+
```

Comment:
Instead of introducing this function why don't you push the implementation of `Archive::getMetadataItem()` into `FileImpl` and avoid the code duplication?

Reply:
Because I don't want to use `Entry`/`Item` in FileImpl.